lib/sequelize.js

```diff
@@ -529,11 +529,12 @@ class Sequelize {
     }).then(connection => {
       const query = new this.dialect.Query(connection, this, options);
 
-      return retry(() => query.run(sql, bindParameters).finally(() => {
-        if (!options.transaction) {
-          return this.connectionManager.releaseConnection(connection);
-        }
-      }), Utils._.extend(this.options.retry, options.retry || {}));
+      return retry(() => query.run(sql, bindParameters), Utils._.extend(this.options.retry, options.retry || {}))
+        .finally(() => {
```

Comment:
You can move this to `finally` block below

Reply:
But `connection` is not defined in that `finally` block below.

Reply:
Right :), then just add a unit test and it should be ready to go